jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml

```diff
@@ -310,6 +310,9 @@
           status-add-test-results: false
           trigger-phrase: 'verify|test'
           cmd: |
+            # verify-godeps compares against upstream, but remote/master might be stale
+            # if .git was retained between runs. Update it explicitly here.
+            git fetch remote master:refs/remotes/remote/master
```

Comment:
It's updating refs/heads/* to track refs/remotes/upstream/*. This updates master, but not remotes/master.

I don't see an option in the scm/git section to make it fetch an unrelated branch.